File: apps/flutter/lib/core/reader_presentation.dart
class ReaderSidebarRowState {
  const ReaderSidebarRowState({
    required this.id,
    required this.title,
    required this.iconName,
    this.subtitle,
    this.badgeCount,
    this.isSelected = false,
  });

  final String id;
  final String title;
  final String? subtitle;
  final String iconName;
  final int? badgeCount;
  final bool isSelected;
}

class ReaderSidebarSectionState {
  const ReaderSidebarSectionState({
    required this.id,
    required this.title,
    required this.rows,
  });

  final String id;
  final String title;
  final List<ReaderSidebarRowState> rows;
}

class ReaderDetailPaneState {
  const ReaderDetailPaneState({
    required this.title,
    required this.metadata,
    this.subtitle,
    this.bodyPreview,
  });

  final String title;
  final String? subtitle;
  final String? bodyPreview;
  final List<String> metadata;
}

class ReaderScreenState {
  const ReaderScreenState({
    required this.headerTitle,
    required this.headerSubtitle,
    required this.sidebarSections,
    required this.isLoading,
    required this.errorMessage,
    this.detailPane,
    this.syncStatusLine,
  });

  final String headerTitle;
  final String headerSubtitle;
  final List<ReaderSidebarSectionState> sidebarSections;
  final ReaderDetailPaneState? detailPane;
  final bool isLoading;
  final String? errorMessage;
  final String? syncStatusLine;
}

File: apps/apple/Shared/ReaderPresentation.swift
import Foundation

public struct ReaderSidebarRowState: Identifiable, Equatable, Sendable {
    public let id: String
    public let title: String
    public let subtitle: String?
    public let iconName: String
    public let badgeCount: Int?
    public let isSelected: Bool

    public init(
        id: String,
        title: String,
        subtitle: String? = nil,
        iconName: String,
        badgeCount: Int? = nil,
        isSelected: Bool = false
    ) {
        self.id = id
        self.title = title
        self.subtitle = subtitle
        self.iconName = iconName
        self.badgeCount = badgeCount
        self.isSelected = isSelected
    }
}

public struct ReaderSidebarSectionState: Identifiable, Equatable, Sendable {
    public let id: String
    public let title: String
    public let rows: [ReaderSidebarRowState]

    public init(id: String, title: String, rows: [ReaderSidebarRowState]) {
        self.id = id
        self.title = title
        self.rows = rows
    }
}

public struct ReaderDetailPaneState: Equatable, Sendable {
    public let title: String
    public let subtitle: String?
    public let bodyPreview: String?
    public let metadata: [String]

    public init(
        title: String,
        subtitle: String? = nil,
        bodyPreview: String? = nil,
        metadata: [String] = []
    ) {
        self.title = title
        self.subtitle = subtitle
        self.bodyPreview = bodyPreview
        self.metadata = metadata
    }
}

public struct ReaderScreenState: Equatable, Sendable {
    public let headerTitle: String
    public let headerSubtitle: String
    public let sidebarSections: [ReaderSidebarSectionState]
    public let detailPane: ReaderDetailPaneState?
    public let isLoading: Bool
    public let errorMessage: String?
    public let syncStatusLine: String?

    public init(
        headerTitle: String,
        headerSubtitle: String,
        sidebarSections: [ReaderSidebarSectionState],
        detailPane: ReaderDetailPaneState?,
        isLoading: Bool,
        errorMessage: String?,
        syncStatusLine: String?
    ) {
        self.headerTitle = headerTitle
        self.headerSubtitle = headerSubtitle
        self.sidebarSections = sidebarSections
        self.detailPane = detailPane
        self.isLoading = isLoading
        self.errorMessage = errorMessage
        self.syncStatusLine = syncStatusLine
    }
}
